amino.py

```diff
@@ -327,19 +292,20 @@ def get_matrix(ops, mut):
         
     """
     
-    dist_mat = np.zeros((len(ops),len(ops)))
-    pairs = np.argwhere(dist_mat == 0)
+    from itertools import combinations_with_replacement
     
+    n = len(ops)
+    dist_mat = np.zeros((n, n))
     
-
-    for pair in pairs:
-        mi = mut.distance(ops[pair[0]], ops[pair[1]])
-        dist_mat[pair[0],pair[1]] = mi
+    for i, j in combinations_with_replacement(range(n), 2):
+        dist = mut.distance(ops[i], ops[j])
+        dist_mat[i, j] = dist
+        dist_mat[j, i] = dist
         
     return dist_mat
 
 # Running clustering on `ops` starting with `seeds`
-def cluster(ops, seeds, mut):
+def cluster(ops: list[OrderParameter], seeds: list[OrderParameter], mut: Memoizer) -> list[OrderParameter]:
```

Comment:
I like this change, should this also get a unit test on a small synthetic data set to make sure we aren't breaking anything?